pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperKitCreationService.java

```diff
@@ -0,0 +1,276 @@
+                                              String ddpLabel, String userId, SimpleResult transactionResults) {
+        String collaboratorSampleId = null;
+        String bspCollaboratorSampleType = kitTypeName;
+        String addressId = null;
+        try {
+            Address address = easyPostUtil.getAddress(kit.getEasypostAddressId());
+            if (address != null) {
+                addressId = address.getId();
+            }
+        } catch (EasyPostException e) {
+            throw new RuntimeException("EasyPost addressId could not be received ", e);
+        }
+
+        if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+            try {
+                // collaborator sample id is the id of the physical sample that GP will work with.
+                collaboratorSampleId =
+                        KitRequestShipping.generateBspSampleID(conn, collaboratorParticipantId, bspCollaboratorSampleType, kitTypeId);
+                KitRequestShipping.writeRequest(ddpInstance.getDdpInstanceId(), juniperKitRequestId, kitTypeId,
+                        kit.getParticipantId().trim(),
+                        collaboratorParticipantId, collaboratorSampleId, userId, addressId, errorMessage, externalOrderNumber, false,
+                        null, ddpInstance, bspCollaboratorSampleType, ddpLabel);
+                //             Not needed now
+                //            kit.setExternalOrderNumber(externalOrderNumber);
+            } catch (Exception e) {
+                transactionResults.resultException = e;
+            }
+        } else {
+            if (kitRequestSettings.getCollaboratorSampleTypeOverwrite() != null) {
+                bspCollaboratorSampleType = kitRequestSettings.getCollaboratorSampleTypeOverwrite();
+            }
+            if (StringUtils.isNotBlank(collaboratorParticipantId)) {
+                collaboratorSampleId =
+                        KitRequestShipping.generateBspSampleID(conn, collaboratorParticipantId, bspCollaboratorSampleType, kitTypeId);
+                if (collaboratorParticipantId == null) {
+                    errorMessage += "collaboratorParticipantId was too long ";
+                }
+                if (collaboratorSampleId == null) {
+                    errorMessage += "collaboratorSampleId was too long ";
+                }
+            }
+
+            String participantID = kit.getJuniperParticipantID();
+            try {
+                String dsmKitRequestId =
+                        KitRequestShipping.writeRequest(ddpInstance.getDdpInstanceId(), juniperKitRequestId, kitTypeId, participantID,
+                                collaboratorParticipantId, collaboratorSampleId, userId, addressId, errorMessage,
+                                kit.getExternalOrderNumber(),
+                                false,
+                                null, ddpInstance, bspCollaboratorSampleType, ddpLabel);
+                kit.setDdpLabel(juniperKitRequestId);
```

Comment:
This clause is similar to that above, but here it includes `kit.setDdpLabel`. Is that intentional?